Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1159,13 +1159,87 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
     return pageSpaceBounds;
 }
 
+static RetainPtr<PDFAnnotation> findFirstTextAnnotationStartingAtIndex(const RetainPtr<NSArray>& annotations, unsigned startingIndex, AnnotationSearchDirection searchDirection)
```

Comment:
There's a system setting to focus controls that are not just text fields; we should respect that, and focus things like checkboxes when it's enabled.

Reply:
This current implementation matches shipping behavior with that setting toggled on and off. I created a bugzilla to properly support focus rings with this setting on: https://bugs.webkit.org/show_bug.cgi?id=269004